assets/app/scripts/services/limits.js

```diff
@@ -0,0 +1,189 @@
+    //   defaultRequest: "200m",
+    //   maxLimitRequestRatio: "10"
+    //   min: "200m",
+    //   max: "2",
+    // }
+
+    var getEffectiveLimitRange = function(limitRanges, computeResource, resourceType) {
+      var effectiveRange = {};
+      angular.forEach(limitRanges, function(limitRange) {
+        angular.forEach(limitRange.spec.limits, function(limit) {
+          if (limit.type !== resourceType) {
+            return;
+          }
+
+          // Find the largest minimum.
+          if (limit.min && isLarger(limit.min[computeResource], effectiveRange.min)) {
+            effectiveRange.min = limit.min[computeResource];
+          }
+          // Find the smallest maximum.
+          if (limit.max && isSmaller(limit.max[computeResource], effectiveRange.max)) {
+            effectiveRange.max = limit.max[computeResource];
+          }
+          // Find default limit and request.
+          // TODO: Decide the correct behavior when there is more than one
+          //       limit range with different defaults set.
+          if (limit['default']) {
+            effectiveRange.defaultLimit = limit['default'][computeResource] || effectiveRange.defaultLimit;
+          }
+          if (limit.defaultRequest) {
+            effectiveRange.defaultRequest = limit.defaultRequest[computeResource] || effectiveRange.defaultRequest;
+          }
+          // Find the smallest max limit/request ratio.
+          var maxRatio;
+          if (limit.maxLimitRequestRatio) {
+            maxRatio = limit.maxLimitRequestRatio[computeResource];
+            if (maxRatio && (!effectiveRange.maxLimitRequestRatio ||
+                             maxRatio < effectiveRange.maxLimitRequestRatio)) {
+              effectiveRange.maxLimitRequestRatio = maxRatio;
+            }
+          }
+        });
+      });
+
+      return effectiveRange;
+    };
+
+    // Tests that the total request and total limit for all containers is
+    // within the limit range minimum and maximum, respectively, for the pod.
+    // Returns an array of error messages, or an empty array if no problems.
+    var validatePodLimits = function(limitRanges, computeResource, containers) {
+      if (!containers || !containers.length) {
+        return [];
+      }
+
+      var podLimits = getEffectiveLimitRange(limitRanges, computeResource, 'Pod');
+      var containerLimits = getEffectiveLimitRange(limitRanges, computeResource, 'Container');
+
+      // Use usageValue to normalize units.
+      var requestTotal = 0,
+          limitTotal = 0,
+          min = podLimits.min && usageValue(podLimits.min),
+          max = podLimits.max && usageValue(podLimits.max),
+          problems = [],
+          computeResourceLabel = $filter('computeResourceLabel')(computeResource, true);
+
+      // Total the usage for all containers.
+      angular.forEach(containers, function(container) {
+        var resources = container.resources;
+        if (!resources) {
+          return;
```

Comment:
wouldn't this container's defaults still matter, you wouldn't have validation on these fields but the defaults should still go against the total